### ISSUE TITLE
Can you go back to android after flashing u-boot?

### ISSUE THREAD

**Issue by djkido316** [xc-racer99/u-boot-aries#36]
I just found a captivate inside my drawer which has been there for 9 years :D

Well it seems like a noob question but haven't tried to flash u-boot yet hence i'm asking can i go back to android after flashing the u-boot or you cannot go back to android at all btw i would be flashing u-boot to kernel partition not to stock IBL.

One more thing i would like to ask in case i decided to wipe  Onenand mentioned here to simulate a Unbrickable mod mentioned [Here](https://github.com/xc-racer99/u-boot-aries/issues/3#issue-336344661) can i flash stock android using heimdall ?

**Comment by xc-racer99** [comment]
> Well it seems like a noob question but haven't tried to flash u-boot yet hence i'm asking can i go back to android after flashing the u-boot or you cannot go back to android at all btw i would be flashing u-boot to kernel partition not to stock IBL.

Yep, you'd just do a normal flash using Heimdall.  Basically all you're doing is replacing the "kernel" with u-boot, which in turn can then load a kernel.

Note that the mainline kernel port done has not been tested on the captivate - nor has this u-boot.  So, beware :)



> One more thing i would like to ask in case i decided to wipe Onenand mentioned here to simulate a Unbrickable mod mentioned [Here](https://github.com/xc-racer99/u-boot-aries/issues/3#issue-336344661) can i flash stock android using heimdall ?

No, you'd need to use the unbrickable recovery - ie follow the directions from https://xdaforums.com/t/how-to-unbrick-the-unbrickable-captivate.1242466/  

If you do try this method, make absolutely sure you have a way of restoring your efs partition in BML (ie a nandroid from an MTD ROM will not work on a BML ROM, although if you don't care about that, go ahead).  It will completely erase absolutely everything from the oneNAND.

**Comment by djkido316** [comment]
> > Well it seems like a noob question but haven't tried to flash u-boot yet hence i'm asking can i go back to android after flashing the u-boot or you cannot go back to android at all btw i would be flashing u-boot to kernel partition not to stock IBL.
> 
> Yep, you'd just do a normal flash using Heimdall. Basically all you're doing is replacing the "kernel" with u-boot, which in turn can then load a kernel.
> 
> Note that the mainline kernel port done has not been tested on the captivate - nor has this u-boot. So, beware :)
> 
> > One more thing i would like to ask in case i decided to wipe Onenand mentioned here to simulate a Unbrickable mod mentioned [Here](https://github.com/xc-racer99/u-boot-aries/issues/3#issue-336344661) can i flash stock android using heimdall ?
> 
> No, you'd need to use the unbrickable recovery - ie follow the directions from https://xdaforums.com/t/how-to-unbrick-the-unbrickable-captivate.1242466/
> 
> If you do try this method, make absolutely sure you have a way of restoring your efs partition in BML (ie a nandroid from an MTD ROM will not work on a BML ROM, although if you don't care about that, go ahead). It will completely erase absolutely everything from the oneNAND.

Thank you for replying, Well the wiki was a bit confusing for someone like me, Anyway i'll compile and try to flash the u-boot as kernel later tonight, Will try to boot mainline kernel with it, I actually really wanted to try PMOS since aries devices are really compatible with each other i'm even using your AOSP 7.1 port for I9000 in cappy and except for the swapped gpio its working great :D.

Also will try to simulate unbrickable mod first hand since i don't care about EFS that much.

**Comment by xc-racer99** [comment]
> Thank you for replying, Well the wiki was a bit confusing for someone like me, Anyway i'll compile and try to flash the u-boot as kernel later tonight, Will try to boot mainline kernel with it, I actually really wanted to try PMOS since aries devices are really compatible with each other i'm even using your AOSP 7.1 port for I9000 in cappy and except for the swapped gpio its working great :D.

Yeah, likely, but there could be some issues.  Note I haven't touch anything in a few years, so it might not be compatible with the latest pmbootstrap changes.



> Also will try to simulate unbrickable mod first hand since i don't care about EFS that much.

Note this is just a temporary unbrickable mod - the moment you flash the first stage bootloader, it is gone.  For EFS, from an MTD ROM, there is usually a backup someone on the SD card called efs.tar, in an efs folder maybe?  It's been a long time...

**Comment by djkido316** [comment]
Well i just tried u-boot using s5p_aries_defconfig, u-boot well boots but no hardware keys work.
